Shared/Tests.Sync.Shared/MergeByPKTests.cs

```diff
@@ -0,0 +1,122 @@
+﻿////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2016 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+using System;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+using IntegrationTests;
+using NUnit.Framework;
+using Realms;
+using Realms.Sync;
+
+namespace Tests.Sync
+{
+    [TestFixture, Preserve(AllMembers = true)]
+    public class MergeByPKTests
+    {
+        [NUnit.Framework.Explicit]
+        [TestCaseSource(nameof(MergeTestCases))]
+        public async void WhenObjectHasPK_ShouldMergeCorrectly(Type objectType, object pkValue, Func<dynamic, bool> pkValueChecker)
+        {
+            User.ConfigurePersistence(UserPersistenceMode.NotEncrypted);
```

Comment:
I feel we should do this for all our tests (at least temporarily) to avoid having 1 failing test each time. I was planning to do it in a separate PR though.